### DISCUSSION TITLE
How do I set up the zksync-era and  block-explorer locally? [ZK Stack] - <>

### DISCUSSION THREAD

**Discussion by peter-22-81** [zkSync-Community-Hub/zksync-developers#270] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [X] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [X] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.73.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I want to set up a development environment for zkSync-era on my own server, including a block-explorer. The data in the block-explorer should be read from the database on myself server.

### Expected Behavior

![1705031780455](https://github.com/zkSync-Community-Hub/zksync-developers/assets/110380583/8175c04e-3d58-4025-b914-0f5f65b99430)

### Current Behavior

![1705031723733](https://github.com/zkSync-Community-Hub/zksync-developers/assets/110380583/9556bb31-0142-436b-b8dc-51d8a7afef82)

http://localhost:3010/?network=local

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by SuccinctPaul** [comment]
Meet the same problem too!!!

* My Run steps:
* clone https://github.com/matter-labs/block-explorer
* docker-compose up

**Reply by vasyl-ivanchuk** [reply]
hi @ChengYueJia 👋 , thanks for creating an issue https://github.com/matter-labs/block-explorer/issues/151 for this problem. We will look into it.

**Comment by vasyl-ivanchuk** [comment]
Hi @peter-22-81 👋 , thanks for reporting. We're looking into the problem, please track the progress in the issue @ChengYueJia created: https://github.com/matter-labs/block-explorer/issues/151.

**Reply by vasyl-ivanchuk** [reply, marked as answer]
@peter-22-81 @ChengYueJia could you please pull the latest from the [Block Explorer repo](https://github.com/matter-labs/block-explorer) and give it a try? There were a few updates to the compose recently, and the latest version works fine for me on Ubuntu."

**Reply by SuccinctPaul** [reply]
Yeah, no problem. I'll have a try.

**Reply by SuccinctPaul** [reply]
Hi, @vasyl-ivanchuk, It works. Thx a lot.

**Reply by vasyl-ivanchuk** [reply]
great, thanks for checking! 🎉

**Comment by kahkaha54** [comment]
Fixed fees regardless of gas price. nice idea

**Comment by upgrade1m** [comment]
Hey ! you can try out foundry-zksync here: https://github.com/matter-labs/foundry-zksync

**Comment by knowalosbra** [comment]
It seems like the issue you're encountering involves discrepancies in timestamps between the expected behavior and the current behavior of your development environment for zkSync-era. To address this, you may need to ensure that your server's system clock is synchronized accurately and that the block explorer is configured to read data from the correct database on your server. Additionally, verifying the configurations related to timestamp handling within your development environment could help in resolving this issue.

**Comment by kaycogrisitbran** [comment]
Ensure that the block explorer is configured to read data from the local database on your server. Verify the connection settings and database credentials to ensure proper data retrieval.

**Comment by amandakob** [comment]
To set up a development environment for zkSync-era on your own server with a block explorer reading data from your local database, follow the installation instructions provided in the respective repositories of the system contracts, zkEVM, server, prover, fee withdrawer, block explorer, and portal. Ensure compatibility between the Rust version and the operating system you're using. Utilize the provided documentation and resources to troubleshoot any issues encountered during the setup process.

**Comment by Amyevesky** [comment]
I'm experiencing the same issue when setting up zkStack testnest ecosystem on my AWS server. The installation appears to complete successfully, but the block explorer isn't displaying any data. Has anyone found a solution or workaround for this?


![Screenshot from 2025-06-10 16-54-51](https://github.com/user-attachments/assets/8b750cf5-b561-4c20-9544-ef579591a5d6)
